sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeContinuation.java

```diff
@@ -34,25 +49,77 @@ public FeedRangeInternal getFeedRange() {
         return this.feedRange;
     }
 
-    public abstract String getContinuation();
+    public abstract CompositeContinuationToken getCurrentContinuationToken();
 
     public abstract void replaceContinuation(String continuationToken);
 
     public abstract boolean isDone();
 
     public abstract void validateContainer(String containerRid);
 
-    /* TODO fabianm - infinite recursion
-    public static FeedRangeContinuation tryParse(String toStringValue) {
-        return FeedRangeCompositeContinuationImpl.tryParse(toStringValue);
-    }*/
+    public void populatePropertyBag() {
+        super.populatePropertyBag();
+    }
 
-    public abstract ShouldRetryResult handleChangeFeedNotModified(
-        RxDocumentServiceResponse responseMessage);
+    public static FeedRangeContinuation tryParse(String jsonString) {
+        if (jsonString == null) {
+            return null;
+        }
+
+        try
+        {
+            return FeedRangeCompositeContinuationImpl.parse(jsonString);
+        }
+        catch (final IOException ioError) {
+            LOGGER.debug(
+                "Failed to parse feed range continuation JSON {}",
+                jsonString,
+                ioError);
+            return null;
+        }
+    }
+
+    public static FeedRangeContinuation convert(final String continuationToken) {
+        if (Strings.isNullOrWhiteSpace(continuationToken)) {
+            throw new NullPointerException("continuationToken");
+        }
+
+        FeedRangeContinuation returnValue = tryParse(continuationToken);
+        if (returnValue != null) {
+            return returnValue;
+        }
+
+        throw new IllegalArgumentException(
+            String.format(
+                "Invalid Feed range continuation token '%s'",
+                continuationToken));
+    }
+
+    public static FeedRangeContinuation createForFullFeedRange(
+        String containerRid,
+        FeedRangeInternal feedRange) {
+
+        checkNotNull(containerRid, "Argument 'collectionLink' must not be null");
+        checkNotNull(feedRange, "Argument 'feedRange' must not be null");
+
+        List<Range<String>> ranges = new ArrayList<>();
+        ranges.add(
+            new Range<>(
+                PartitionKeyInternalHelper.MinimumInclusiveEffectivePartitionKey,
```

Comment:
Why would the ranges have the full range? What if the `feedRange` is smaller?

Reply:
Fixed